test-network/network.sh

```diff
@@ -19,27 +19,19 @@ export VERBOSE=false
 . scripts/utils.sh
 
 # Obtain CONTAINER_IDS and remove them
-# TODO Might want to make this optional - could clear other containers
 # This function is called when you bring a network down
 function clearContainers() {
-  CONTAINER_IDS=$(docker ps -a | awk '($2 ~ /dev-peer.*/) {print $1}')
-  if [ -z "$CONTAINER_IDS" -o "$CONTAINER_IDS" == " " ]; then
-    infoln "No containers available for deletion"
-  else
-    docker rm -f $CONTAINER_IDS
-  fi
+  infoln "Removing remaining containers"
+  docker rm -f $(docker ps -aq --filter label=service=hyperledger-fabric) 2>/dev/null || true
+  docker rm -f $(docker ps -aq --filter name='dev-peer*') 2>/dev/null || true
 }
 
 # Delete any images that were generated as a part of this setup
 # specifically the following images are often left behind:
 # This function is called when you bring the network down
 function removeUnwantedImages() {
-  DOCKER_IMAGE_IDS=$(docker images | awk '($1 ~ /dev-peer.*/) {print $3}')
-  if [ -z "$DOCKER_IMAGE_IDS" -o "$DOCKER_IMAGE_IDS" == " " ]; then
-    infoln "No images available for deletion"
-  else
-    docker rmi -f $DOCKER_IMAGE_IDS
-  fi
+  infoln "Removing generated chaincode docker images"
+  docker image rm -f $(docker images -aq --filter reference='dev-peer*') 2>/dev/null || true
```

Comment:
this should be `rmi` rather than `rm` right?

Reply:
Just to close this for whoever might visit this in the future, `--filter` is not available on `rmi` which is why I used `image rm`